hummingbot/connector/exchange/bloxroute_openbook/bloxroute_openbook_exchange.py

```diff
@@ -0,0 +1,533 @@
+)
+from hummingbot.connector.exchange.bloxroute_openbook.bloxroute_openbook_auth import BloxrouteOpenbookAuth
+from hummingbot.connector.exchange.bloxroute_openbook.bloxroute_openbook_constants import OPENBOOK_PROJECT
+from hummingbot.connector.exchange.bloxroute_openbook.bloxroute_openbook_order_book import BloxrouteOpenbookOrderBook
+from hummingbot.connector.exchange.bloxroute_openbook.bloxroute_openbook_order_data_manager import (
+    BloxrouteOpenbookOrderDataManager,
+)
+from hummingbot.connector.exchange_py_base import ExchangePyBase
+from hummingbot.connector.trading_rule import TradingRule
+from hummingbot.core.data_type.common import OrderType, TradeType
+from hummingbot.core.data_type.in_flight_order import InFlightOrder, OrderState, OrderUpdate, TradeUpdate
+from hummingbot.core.data_type.order_book import OrderBook, OrderBookMessage
+from hummingbot.core.data_type.order_book_tracker_data_source import OrderBookTrackerDataSource
+from hummingbot.core.data_type.trade_fee import AddedToCostTradeFee, DeductedFromReturnsTradeFee, TradeFeeBase
+from hummingbot.core.data_type.user_stream_tracker_data_source import UserStreamTrackerDataSource
+from hummingbot.core.network_iterator import NetworkStatus
+from hummingbot.core.web_assistant.web_assistants_factory import WebAssistantsFactory
+
+if TYPE_CHECKING:
+    from hummingbot.client.config.config_helpers import ClientConfigAdapter
+
+s_logger = None
+
+
+class BloxrouteOpenbookExchange(ExchangePyBase):
+    """
+    BloxrouteOpenbookExchange connects with BloxRoute Labs Solana Trader API provides order book pricing, user account tracking and
+    trading functionality.
+    """
+
+    web_utils = web_utils
+
+    def __init__(
+        self,
+        client_config_map: "ClientConfigAdapter",
+        bloxroute_api_key: str,
+        solana_wallet_public_key: str,
+        solana_wallet_private_key: str,
+        open_orders_address: str,
+        trading_pairs: Optional[List[str]] = None,
+        trading_required: bool = True,
+    ):
+        """
+        :param bloxroute_api_key: The bloxRoute Labs authorization header to connect with solana trader api
+        :param solana_wallet_private_key: The secret key for a solana wallet
+        :param trading_pairs: The market trading pairs which to track order book data.
+        :param trading_required: Whether actual trading is needed.
+        """
+
+        self.logger().exception("creating blox route exchange")
+        self.logger().exception("api key is " + bloxroute_api_key)
+        self.logger().exception("pub key is " + solana_wallet_public_key)
+        self.logger().exception("private key is " + solana_wallet_private_key)
+        self.logger().exception("open orders address is " + open_orders_address)
+
+        self._auth_header = "YmUwMjRkZjYtNGJmMy00MDY0LWE4MzAtNjU4MGM3ODhkM2E4OmY1ZWVhZTgxZjcwMzE5NjQ0ZmM3ZDYwNmIxZjg1YTUz"
+        self._sol_wallet_public_key = solana_wallet_public_key
```

Comment:
BTW public can be derived from public key, so the user only needs to provide the public key.

From Python SDK:

```python
    pk = load_private_key(pkey_str)
    pk.public_key
```